### PR TITLE
Update documentation on Azure Setup

### PR DESCRIPTION
**Pull request by marco-lancini**
I've recently tried to setup the Azure ingestion, and I've found the doc lacking of details.

This PR extends the documentation with a step by step approach for running the Azure ingestion

### CODE REVIEW

**Comment by mpurusottamc** [comment]
Looks good.

File: docs/setup/config/azure.md [hunk 1]
```diff
@@ -3,20 +3,22 @@
...
+    * Authenticate: `$ az login`
+    * Create a Service Principal: `$ az ad sp create-for-rbac --name cartography --role Reader`
+    * Note the values of the `tenant`, `appId`, and `password` fields
+1. Populate environment variables with the values generated in the previous step (e.g., `AZURE_TENANT_ID`, `AZURE_CLIENT_ID`, `AZURE_CLIENT_SECRET`)
+1. Call the `cartography` CLI with:
+    ```bash
+    --azure-sp-auth --azure-sync-all-subscriptions      \
+    --azure-tenant-id ${AZURE_TENANT_ID}                \
+    --azure-client-id ${AZURE_CLIENT_ID}                \
+    --azure-client-secret-env-var AZURE_CLIENT_SECRET
```
Comment (kunaals) [inline docs/setup/config/azure.md:23]: nit
```suggestion
    --azure-client-secret-env-var ${AZURE_CLIENT_SECRET}
```

**Review by kunaals — APPROVED** [review]
left one small nit, otherwise looks good

File: docs/setup/config/azure.md (hunk 1 above)
Comment (marco-lancini) [reply docs/setup/config/azure.md:23]: > nit
> 
> ```suggestion
> 
>     --azure-client-secret-env-var ${AZURE_CLIENT_SECRET}
> 
> ```

That's actually ok purpose, as Cartography expects the name of an environment variable for the secret, not the secret itself

**Pull request merged**